### PR TITLE
[WIP] Protobuf support

### PR DESCRIPTION
**Pull request by zhkvia**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/src/DataStreams/ProtobufRowInputStream.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    ReadBuffer & istr_, const Block & header_,
+    const String & schema_dir, const String & schema_file, const String & root_obj
+) : istr(istr_), header(header_)
+{
+    String schema_path = getSchemaPath(schema_dir, schema_file);
+    // TODO: remove
+    std::cerr << "schema_path: " << schema_path << std::endl;
+    std::cerr << "root obj: " << root_obj << std::endl;
+
+    int fd = open(schema_path.c_str(), O_RDONLY);
```
Comment (ludv1x) [inline dbms/src/DataStreams/ProtobufRowInputStream.cpp]: It is better to use high-level interface.
If an exception is thrown below, the open file will remain.

**Comment by alexey-milovidov** [comment]
It is 5% done, at most. This PR has no future.
We will continue in a separate PR.

**Comment by knoguchi** [comment]
I know this is a closed PR but I tested anyway.   The biggest problem is it can't handle multiple PB messages.  The delimiter option in Kafka storage won't work because PB is not a delimiter safe format unfortunately.   One idea is to frame the PB payload with a run length prefix.  Can we introduce such option to Kafka storage?

**Comment by alexey-milovidov** [comment]
We have started another task as a replacement for this PR.
Protobuf messages will be delimited by size, as in https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/util/delimited_message_util.h

**Comment by alexey-milovidov** [comment]
If your messages in Kafka will be prefixed by size in appropriate format (varint), the Kafka storage will be able to read them from Protobuf format.

**Comment by alexey-milovidov** [comment]
Short task description in russian: https://gist.githubusercontent.com/alexey-milovidov/5551a7c7b7aba731a0a74f01972ddc69/raw/054237b5165a26a6482329c1416ef153bec37b8e/protobuf_task_ru.txt

**Comment by knoguchi** [comment]
Thanks for the description (I've read it using Google Translator).  It seems Protobuf v3.3.0 added the delimited message utilities for C++ (Protobuf for Java supports delimiter for a long time).   

I did a bit of experiment.  Unfortunately `ParseDelimitedFromZeroCopyStream` takes `MessageLite` instead of `Message`.  MessageLite lacks reflection methods that are necessary to work with DynamicMessage.  There might be a way to work around but probably it's ok to copy some functions from the utility to ClickHouse while keeping the compatibility with the Google's official delimiters.

By the way I've written "jq for Protobuf".   It can query nested Protobuf messages just like jq for JSON.   You mentioned in the document that arrays, nested messages and other complex types won't be supported.  The jq can flatten the complex structure.  I'm integrating it to ClickHouse.

**Pull request closed without merging**